### PR TITLE
Generate sitemaps individually

### PR DESCRIPTION
**Pull request by bilbof**
Generating and writing the XML files to temp files individually, rather than generating them all at once and then writing them to temp files, is less memory intensive.

https://trello.com/c/tGJcHr2f/1041-investigate-sitemap-generation-failures

### CODE REVIEW

File: lib/sitemap/sitemap_generator.rb [hunk 1]
```diff
@@ -1,15 +1,35 @@
...
+      end
+    end
+
+    builder.to_xml
+  end
+
+private
+
+  EXCLUDED_FORMATS = %w[recommended-link].freeze
+  SITEMAP_LIMIT = 25_000
```
Comment (barrucadu) [inline lib/sitemap/sitemap_generator.rb:31]: We may be able to up this to 50,000 now - which is the max amount of entries one sitemap file can have.  It used to be 50,000, but when we switched to ES5 the rake task was regularly running out of memory so I halved it.

**Review by barrucadu — APPROVED** [review]

**Review by hannako — APPROVED** [review]
👍

File: lib/sitemap/sitemap_generator.rb (hunk 1 above)
Comment (bilbof) [reply lib/sitemap/sitemap_generator.rb:31]: That's interesting, is there a benefit to having fewer files?

Comment (barrucadu) [reply lib/sitemap/sitemap_generator.rb:31]: Not sure.  I imagine having 50,000 1-entry files would be bad, but I don't know if 1 50,000-entry file is better than 2 25,000-entry files.

**Pull request merged**